### DISCUSSION TITLE
Issue with the usage of `=` vs `:` for options in Quarto code chunks

### DISCUSSION THREAD

**Discussion by JosephBARBIERDARNAL** [quarto-dev/quarto-cli#11574] [Q&A] [answered]
### Bug description

**Description:**
I am encountering an issue when using options in Quarto code chunks. Specifically, there is a problem with the assignment of options when using the `=` sign compared to the `:` sign. 

### Steps to Reproduce:
1. In a Quarto code chunk, I use the `=` operator to assign options:
   ```{r main table}
   #| echo = FALSE
   #| column = screen
   #| out-width = 100%
   ```

```
Error in parse(text = code, keep.source = FALSE) : 
  <text>:2:1: unexpected symbol
1: alist( column = screen
2: out
   ^
Calls: .main ... csv_options -> handle_error -> eval -> parse_only -> parse
Invalid syntax for chunk options at lines 33-56 (final.qmd):

column = screen
out-width = 100%
echo = FALSE
```

2. However, if I use the `:` operator instead of `=`, the chunk works fine:
   ```{r main table}
   #| echo : FALSE
   #| column : screen
   #| out-width : 100%
   ```

3. The `=` operator only works correctly when the `echo` option is alone:
   ```{r main table}
   #| echo = FALSE
   ```
   But as soon as I try to add other options (like `column` or `out-width`), using `=` breaks the chunk.
  

  

### Steps to reproduce

```qmd
---
title: "Quarto Bug Report"
format: html
---

## Quarto Bug Report - `=` vs `:`

### Example with `=` (fails)

```{r main table}
#| echo = FALSE
#| column = screen
#| out-width = 100%
```

### Expected behavior

- Both `=` and `:` should be functional for assigning options in Quarto code chunks (or just be consistent)

### Actual behavior

The `=` operator only works when `echo` is used alone, while using `=` for other options like `column` or `out-width` causes the chunk to fail. The `:` operator seems to work fine for all options.

### Your environment

- macOS Sonoma 14.6.1

- Positron: 2024.08.0 (Universal) build 48

- Quarto: 1.5.57

### Quarto check output

```bash
Quarto 1.5.57
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.57
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Library/TeX/texbin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.5
      Path: /Users/josephbarbier/Desktop/30daysmapchallenge/venv/bin/python
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
      knitr: 1.49
      rmarkdown: 2.29

[✓] Checking Knitr engine render......OK
```

**Comment by JosephBARBIERDARNAL** [comment]
_Note: I don't know if that's the right repo_

**Comment by mcanouil** [comment, marked as answer]
This is not a bug.
You are mixing knitr syntaxes which you should not do.

All this is described in `knitr`: https://yihui.org/knitr/options/

Quarto uses and recommends YAML, thus there should be nothing in between brackets beside the language and no equal signs anywhere.

Recommended YAML syntax:

````
```{r}
#| label: "my-label"
#| echo: false
#| out-width: 100%
...
```
````

R inline syntax (not recommended in Quarto):

````
```{r my-label, echo = FALSE, out-width = "100%"}
...
```
````

R multiline syntax (not recommended in Quarto):

````
```{r}
#| label = "my-label",
#| echo = FALSE,
#| out-width = "100%"
...
```
````

**Reply by JosephBARBIERDARNAL** [reply]
Okay, that sounds good! I wasn't aware of this.
